pkgs/top-level/all-packages.nix

```diff
@@ -288,13 +288,13 @@ with pkgs;
 
   # gitlab example
   fetchFromGitLab = {
-    owner, repo, rev, domain ? "gitlab.com", name ? "source",
+    owner, repo, rev, domain ? "gitlab.com", name ? "source", group ? null,
     ... # For hash agility
   }@args: fetchzip ({
     inherit name;
-    url = "https://${domain}/api/v4/projects/${owner}%2F${repo}/repository/archive.tar.gz?sha=${rev}";
-    meta.homepage = "https://${domain}/${owner}/${repo}/";
-  } // removeAttrs args [ "domain" "owner" "repo" "rev" ]) // { inherit rev; };
+    url = "https://${domain}/api/v4/projects/${lib.optionalString (group != null) group+"%2F"}${owner}%2F${repo}/repository/archive.tar.gz?sha=${rev}";
+    meta.homepage = "https://${domain}/${lib.optionalString (group != null) group+"/"}${owner}/${repo}/";
+  } // removeAttrs args [ "domain" "owner" "group" "repo" "rev" ]) // { inherit rev; };
```

Comment:
Can you please split this into a separate commit?

Reply:
This breaks `fetchFromGitLab`.

Before:
`https://gitlab.com/api/v4/projects/psmisc%2Fpsmisc/repository/archive.tar.gz?sha=v23.2`
After:
`https://gitlab.com/api/v4/projects/%2Fpsmisc%2Fpsmisc/repository/archive.tar.gz?sha=v23.2`

Reply:
@azlig I'd like to revert this PR until this is resolved, okay?

Reply:
@ryantm: Ah, this is just a grouping issue, going to fix that instead.

Reply:
More context https://github.com/NixOS/nixpkgs/pull/46977#issuecomment-424722471

Reply:
Fixed in 593062ddc2104cf91b1cf9cc0e4f4b1437f59f6b.